wcfsetup/install/files/lib/data/user/UserProfileAction.class.php

```diff
@@ -567,6 +572,82 @@ public function updateSpecialTrophies()
         }
     }
 
+    /**
+     * Sets an avatar for a given user. The given file will be renamed and is gone after this method call.
+     *
+     * @throws UserInputException If none or more than one user is given.
+     * @throws \InvalidArgumentException If the given avatar is not an image or is incorrect sized.
+     */
+    public function setAvatar(): array
+    {
+        $user = $this->getSingleObject();
+
+        $imageData = \getimagesize($this->parameters['fileLocation']);
+
+        if (!$imageData) {
+            throw new \InvalidArgumentException("The given file is not an image.");
+        }
+
+        if ($imageData[0] != UserAvatar::AVATAR_SIZE || $imageData[1] != UserAvatar::AVATAR_SIZE) {
+            throw new \InvalidArgumentException(
+                "The given avatar does not have the size of " . UserAvatar::AVATAR_SIZE . "x" . UserAvatar::AVATAR_SIZE . "."
```

Comment:
```suggestion
                "The given file does not have the size of " . UserAvatar::AVATAR_SIZE . "x" . UserAvatar::AVATAR_SIZE . "."
```

Possibly using `\sprintf()` (`%d` for the integer placeholders) would be more readable?